tests/test_parts_manager_split_tags.py

```diff
@@ -0,0 +1,573 @@
+"""Tests for split thinking tag handling in ModelResponsePartsManager."""
+
+from inline_snapshot import snapshot
+
+from pydantic_ai._parts_manager import ModelResponsePartsManager
+from pydantic_ai.messages import (
+    PartDeltaEvent,
+    PartStartEvent,
+    TextPart,
+    TextPartDelta,
+    ThinkingPart,
+    ThinkingPartDelta,
+)
+
+
+def test_handle_text_deltas_with_split_think_tags_at_chunk_start():
+    """Test split thinking tags when tag starts at position 0 of chunk."""
+    manager = ModelResponsePartsManager()
+    thinking_tags = ('<think>', '</think>')
+
+    # Chunk 1: "<thi" - starts at position 0, buffer it  # codespell:ignore thi
+    events = list(manager.handle_text_delta(vendor_part_id='content', content='<thi', thinking_tags=thinking_tags))
+    assert len(events) == 0  # Buffered, no events yet
+    assert manager.get_parts() == []
+
+    # Chunk 2: "nk>" - completes the tag
+    events = list(manager.handle_text_delta(vendor_part_id='content', content='nk>', thinking_tags=thinking_tags))
+    assert len(events) == 1
+    assert events[0] == snapshot(
+        PartStartEvent(index=0, part=ThinkingPart(content='', part_kind='thinking'), event_kind='part_start')
```

Comment:
Since we're doing more buffering now, could we make it so that we don't yield a `ThinkingPart` with `content=''` and only yield once we see actual text? I did it like this before because we were checking the most recent part to see if it was a `ThinkingPart`, to determine whether we're in "thinking mode", but we can now keep track of that more directly